### PR TITLE
Join bail if kubernetes is already installed

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: scripts/common/kubernetes.sh [hunk 1]
```diff
@@ -549,3 +549,31 @@ function kubernetes_pod_completed() {
...
+}
+
+function kubernetes_is_join_node() {
+    if cat /opt/replicated/kubeadm.conf 2>/dev/null | grep -q 'kind: JoinConfiguration'; then
+        return 0
+    fi
+    return 1
+}
+
+function kubernetes_is_installed() {
```
Comment (emosbaugh) [inline scripts/common/kubernetes.sh:568]: is there a better way to detect this?

File: scripts/common/kubernetes.sh [hunk 2]
```diff
@@ -549,3 +549,31 @@ function kubernetes_pod_completed() {
 
     return 1
 }
+
+function kubernetes_is_current_cluster() {
+    local api_service_address="$1"
+    if cat /opt/replicated/kubeadm.conf 2>/dev/null | grep -q "${api_service_address}"; then
```
Comment (areed) [inline scripts/common/kubernetes.sh:555]: It's possible they've changed their load balancer address and run the `set-kubeconfig-server` task. How about checking `/etc/kubernetes/kubelet.conf` instead?

File: scripts/common/kubernetes.sh (hunk 1 above)
Comment (areed) [reply scripts/common/kubernetes.sh:568]: I can't think of one

File: scripts/common/kubernetes.sh (hunk 2 above)
Comment (emosbaugh) [reply scripts/common/kubernetes.sh:555]: what about on secondary nodes? ```$ ls /etc/kubernetes/
manifests  pki```

Comment (emosbaugh) [reply scripts/common/kubernetes.sh:555]: ill check both

**Review by areed — APPROVED** [review]

**Pull request merged**